2000-08-01  Jakub Jelinek  <jakub@redhat.com>

	* combine.c (combine_simplify_rtx): Use op0_mode and only
	try other modes if op0_mode is VOIDmode.

--- gcc/combine.c.jj	Mon Jul 31 20:02:09 2000
+++ gcc/combine.c	Tue Aug  1 14:31:43 2000
@@ -3640,7 +3640,9 @@ combine_simplify_rtx (x, op0_mode, last,
       break;
     case '<':
       {
-	enum machine_mode cmp_mode = GET_MODE (XEXP (x, 0));
+	enum machine_mode cmp_mode =
+	  op0_mode == VOIDmode ? GET_MODE (XEXP (x, 0)) : op0_mode;
+
 	if (cmp_mode == VOIDmode)
 	  cmp_mode = GET_MODE (XEXP (x, 1));
 	temp = simplify_relational_operation (code, cmp_mode,
